### PR TITLE
Allow passing a custom Sorbet bin

### PR DESCRIPTION
**Pull request by Morriar**
Add the option `--sorbet SORBET_BIN` to the commands `tc`, `snapshot` and `timeline` so the user can use a freshly compiled Sorbet instead of the one in the bundle.

Also clean a bit the way we execute Sorbet while at it.

### CODE REVIEW

**Review by RyanBrushett — COMMENTED** [review]
I played around with it, works nicely, looks good.
One question is all.

File: lib/spoom/sorbet.rb [hunk 1]
```diff
@@ -11,73 +11,111 @@
...
+      sig do
+        params(
+          arg: String,
+          path: String,
+          capture_err: T::Boolean,
+          sorbet_bin: T.nilable(String)
+        ).returns([String, T::Boolean])
+      end
+      def srb_tc(*arg, path: '.', capture_err: false, sorbet_bin: nil)
+        arg.prepend("tc") unless sorbet_bin
```
[the remark this answers is not in the document]
Comment (Morriar) [reply lib/spoom/sorbet.rb:47]: Good question.

`srb --version ` delegates to `srb tc --version` so we can pass it directly.

But looking closely at how it works I think I can simplify all of this more: https://github.com/sorbet/sorbet/blob/master/gems/sorbet/bin/srb#L129-L140

[the remark this answers is not in the document]
Comment (Morriar) [reply lib/spoom/sorbet.rb:47]: Here, now they all go through `srb_tc`.

[the remark this answers is not in the document]
Comment (RyanBrushett) [reply lib/spoom/sorbet.rb:47]: Ah I understand, neat!

**Review by RyanBrushett — APPROVED** [review]

**Review by KaanOzkan — APPROVED** [review]

**Pull request merged**